hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -978,17 +979,14 @@ private void bindAWSClient(URI name, boolean dtEnabled) throws IOException {
         .withPathStyleAccess(conf.getBoolean(PATH_STYLE_ACCESS, false))
         .withUserAgentSuffix(uaSuffix)
         .withRequesterPays(conf.getBoolean(ALLOW_REQUESTER_PAYS, DEFAULT_ALLOW_REQUESTER_PAYS))
-        .withExecutionInterceptors(auditManager.createExecutionInterceptors());
+        .withExecutionInterceptors(auditManager.createExecutionInterceptors())
+        .withMinimumPartSize(partSize)
+        .withTransferManagerExecutor(unboundedThreadPool);
 
-    s3Client = ReflectionUtils.newInstance(s3ClientFactoryClass, conf)
-        .createS3ClientV2(getUri(),
-            parameters);
-
-    s3AsyncClient = ReflectionUtils.newInstance(s3ClientFactoryClass, conf)
-        .createS3AsyncClient(getUri(),
-            parameters);
-
-    initTransferManager();
+    S3ClientFactory clientFactory = ReflectionUtils.newInstance(s3ClientFactoryClass, conf);
+    s3Client = clientFactory.createS3ClientV2(getUri(), parameters);
+    s3AsyncClient = clientFactory.createS3AsyncClient(getUri(), parameters);
+    transferManager =  clientFactory.createS3TransferManager(getUri(), parameters);
```

Comment:
At the moment, the logic to determine the region is executed 3 times. Maybe we should consider an initialiser method on the factory.